libyul/AST.h

```diff
@@ -38,20 +38,34 @@ using Type = YulString;
 
 struct DebugData
 {
-	explicit DebugData(langutil::SourceLocation _location, std::optional<int64_t> _astID = {}):
-		location(std::move(_location)),
+	explicit DebugData(
+		langutil::SourceLocation _irLocation,
+		langutil::SourceLocation _originalLocation = {},
+		std::optional<int64_t> _astID = {}
+	):
+		irLocation(std::move(_irLocation)),
+		originalLocation(std::move(_originalLocation)),
 		astID(std::move(_astID))
 	{}
 
 	static std::shared_ptr<DebugData const> create(
-		langutil::SourceLocation _location = {},
+		langutil::SourceLocation _irLocation = {},
+		langutil::SourceLocation _originalLocation = {},
 		std::optional<int64_t> _astID = {}
 	)
 	{
-		return std::make_shared<DebugData const>(std::move(_location), std::move(_astID));
+		return std::make_shared<DebugData const>(
+			std::move(_irLocation),
+			std::move(_originalLocation),
+			std::move(_astID)
+		);
 	}
 
-	langutil::SourceLocation location;
+	/// Location in the Yul code.
+	langutil::SourceLocation irLocation;
```

Comment:
I'm actually not sure we should call this `irLocation` - if you use yul in "native" mode, it is not really an IR.

Reply:
So `nativeLocation`?

Reply:
Or maybe just `yulLocation`?

Reply:
I went with `nativeLocation`.